R/likelihood.R

```diff
@@ -78,9 +132,69 @@ likelihood <- function(chains, statistic = c("size", "length"), offspring_dist,
   checkmate::assert_numeric(
     exclude, null.ok = TRUE
   )
+  # likelihood cannot work with an <epichains> object so convert to
+  # <epichains_summary>
+  if (.is_epichains(chains)) {
+    chains <- summary(chains)
+  }
+
+  # Logic:
+  # 1. If stat_threshold is not specified:
+  #
+  # * If the object is an <epichains_summary>, we'll give preference to the
+  # stat_threshold used in the simulation and stored as an attribute in the
+  # object but we'll warn the user.
+  #
+  # * If the object is numeric, we'll use the default of Inf.
+  #
+  # 2. If stat_threshold is specified, we'll give that precedence.
+  if (missing(stat_threshold) && .is_epichains_summary(chains)) {
+    warning(
+      "`stat_threshold` not specified. ",
+      "Using `stat_threshold` = ",
+      .get_stat_threshold(chains),
+      " as used in the simulation.",
+      call. = FALSE
+    )
```

Comment:
I don't see why this would need to produce a warning - instead, do we not need to warn the user if they *do* supply a stat threshold because it'll be overwritten?

Reply:
yes, but we might then want to produce a warning, at least if it's different to the one contained in the epichain?

Reply:
The current implementation ignores `stat_threshold` if specified and the user is warned. This has been documented in `stat_threshold`. See 515527d0b704cc651c7ed471edd375bd3f4f8d72.